### PR TITLE
feat(1115): flipping nfts

### PR DESCRIPTION
**Pull request by KngZhi**
- feat: model of flipping nft
- feat: resolver

### CODE REVIEW

File: src/server-extension/query/nft.ts [hunk 1]
```diff
@@ -1,4 +1,30 @@
 export const passionQuery = `SELECT DISTINCT ne.issuer as id
 FROM nft_entity ne
 WHERE ne.current_owner = $1
-AND ne.current_owner != ne.issuer`
\ No newline at end of file
+AND ne.current_owner != ne.issuer`
+
+export const flippingQuery = `select
```
Comment (vikiival) [inline src/server-extension/query/nft.ts:6]: Please use the correct SQL formating
- KEYWORDS UPPERCASE (SELECT, FROM, GROUP BY)
- and formatiing (use some online formatter)

File: src/server-extension/query/nft.ts [hunk 2]
```diff
@@ -1,4 +1,30 @@
...
+author,
+e.nft_id,
+date,
+current,
+floor_price,
+s.total,
+s.unique_collectors,
+s.emote_count
+order by count desc
+limit 20`
```
Comment (vikiival) [inline src/server-extension/query/nft.ts:30]: make as parameter

**Review by vikiival — CHANGES_REQUESTED** [review]

**Review by vikiival — COMMENTED** [review]
noice

**Comment by KngZhi** [comment]
https://github.com/kodadot/nft-gallery/issues/1115

File: src/server-extension/resolvers/flippingNfts.ts [hunk 3]
```diff
@@ -2,8 +2,14 @@ import { Query, Resolver, Arg } from 'type-graphql'
 import type { EntityManager } from 'typeorm'
 import { NFTEntity } from '../../model/generated'
 import { FlippingNFT } from '../model/nft.model'
-import { flippingQuery } from "../query/nft";
-import { makeQuery } from "../utils";
+import { flippingQuery, previousPriceQuery } from "../query/nft";
+import { makeQuery, toSqlInParams } from "../utils";
+
+type previousNft = {
```
Comment (vikiival) [inline src/server-extension/resolvers/flippingNfts.ts:9]: Type with UpperCase pls

**Review by vikiival — CHANGES_REQUESTED** [review]

**Comment by vikiival** [comment]
As I have nothing specific for this PR. 
I honestly think that this should be cached the same way as spotlight or series.

The computation on the DB is overkill.
I can spawn @eldargab

**Comment by eldargab** [comment]
I'm lacking context here. Generally plain database queries are of course better unless it becomes expensive.
May be  @ma-shulgin can provide a better input.

File: src/server-extension/model/nft.model.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { Field, ObjectType } from 'type-graphql';
+
+@ObjectType()
+export class FlippingNFT {
```
[the remark this answers is not in the document]
Comment (KngZhi) [reply src/server-extension/model/nft.model.ts:4]: Maybe I should change all those `Flipping` to `MostTraded`?

**Comment by vikiival** [comment]
Hey @KngZhi we had some issues with a subsquid node that was unable to process. With your PR I have a feeling that we can buy a ⚰️ for the processor.


What are the next steps: 
- It can't go out as a resolver. 
- so preferrably I would make a caching and a new entity (similar stuff what has been done in #87)

**Comment by KngZhi** [comment]
Hey @vikiival, after I checked with #87 I kind of feel it is unable to do so.

Some fields have relied on the previous query results, there are two sequence queries.

**Comment by KngZhi** [comment]
> Hey @vikiival, after I checked with #87 I kind of feel it is unable to do so.
> 
> Some fields have relied on the previous query results, there are two sequenced queries.

**Comment by petersopko** [comment]
@KngZhi what's the status of this PR? we have now two PRs (this one and https://github.com/kodadot/nft-gallery/pull/3063) which are 3 weeks old and still open 😢

**Comment by vikiival** [comment]
I wont merge this as resolver. 

We are already running out of memory and this is simply overkill

**Comment by petersopko** [comment]
based on @vikiival comments, my understanding here is that this should be done as extension of `cache.ts` not as separate resolvers. let's close this one since it's sitting here for a while, and reopen with extended cache whenever you feel ready to do that @KngZhi

**Pull request closed without merging**